internal/pkg/upload/finalize.go

```diff
@@ -0,0 +1,145 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package upload
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"sort"
+	"strings"
+
+	"github.com/rs/zerolog/log"
+)
+
+func (u *Uploader) Complete(ctx context.Context, id string, transitHash string) (Info, error) {
+	// make sure document is freshly fetched, not cached
+	// so accurate status checking happens
+	info, err := FetchUploadInfo(ctx, u.bulker, id)
+	if err != nil {
+		return info, err
+	}
+
+	/*
+		Verify Upload
+	*/
+
+	// if already done, failed or deleted, exit
+	if !info.StatusCanUpload() {
+		return info, err
```

Comment:
if intention to return no error here then 
```
return info, nil
```
otherwise define error if can't upload?